### PR TITLE
Jensen shannon

### PR DESCRIPTION
**Pull request by AlexDo1**
First implementation of the Jensen-Shannon Divergence.

### CODE REVIEW

File: skinfo/__init__.py [hunk 1]
```diff
@@ -1,2 +1,2 @@
 from .metrics import entropy, conditional_entropy, mutual_information
-from .metrics import joint_entropy, cross_entropy, kullback_leibler
\ No newline at end of file
+from .metrics import joint_entropy, cross_entropy, kullback_leibler, jensen_shannon
```
Comment (mmaelicke) [inline skinfo/__init__.py:2]: we should remove the imports altogether. But that's not a part of this PR

File: skinfo/metrics.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import numpy as np
 
 
-def entropy(x, bins):
+def entropy(x, bins, normalize=False, probabilities=False):
```
Comment (mmaelicke) [inline skinfo/metrics.py:4]: I would rather go for `use_probs` or `as_probs`, because `probabilities` is quite long and at least for me not that clear.

**Comment by mmaelicke** [comment]
I will review this as #5 was changed and merged. Then we can pull the HEAD into this one, cause that will resolve some of my potential comments.

File: skinfo/metrics.py [hunk 3]
```diff
@@ -302,8 +353,73 @@ def kullback_leibler(x, y, bins):
...
 
     """
-    # calculte the cross entropy and unconditioned entopy of y
-    hcross = cross_entropy(x, y, bins)
-    hx = entropy(x, bins)
+    # get the bins, only calculate if bins were not calculated before ([array, array])
+    if type(bins) == list:
+        if len(bins) == 2:
+            if (type(bins[0]), type(bins[1])) == (np.ndarray, np.ndarray):
+                pass
```
Comment (mmaelicke) [inline skinfo/metrics.py:360]: I don't get this. And also please use `isinstance(bins, list)` and maybe even check for list and tuple: `isinstance(bins, (list, tuple))`.

File: skinfo/metrics.py [hunk 4]
```diff
@@ -302,8 +353,73 @@ def kullback_leibler(x, y, bins):
...
+    -----
+
+    The Jensen-Shannon Divergence is based on the Kullback-Leibler Divergence
+    and is calculated as follows:
+
+    .. math::
+
+        JSD(x||y) = 1/2 * D_{KL}(p_{x}||p_{m}) + 1/2 * D_{KL}(p_{y}||p_{m})
+
+        p_{m} = 1/2 * (p_{x} + p_{y})
```
Comment (mmaelicke) [inline skinfo/metrics.py:405]: do we have a reference for this?

File: skinfo/metrics.py [hunk 5]
```diff
@@ -302,8 +353,73 @@ def kullback_leibler(x, y, bins):
...
+    # calculate probabilities
+    px = (hist_x / np.sum(hist_x)) + 1e-15
+    py = (hist_y / np.sum(hist_y)) + 1e-15
+
+    # calculate m
+    pm = 0.5 * (px + py)
 
-    return hcross - hx
\ No newline at end of file
+    return 0.5 * kullback_leibler(px, pm, bins=bins, probabilities=True) + 0.5 * kullback_leibler(py, pm, bins=bins, probabilities=True)
```
Comment (mmaelicke) [inline skinfo/metrics.py:425]: please split this up into calculation, and return. This statement is too long.

File: skinfo/metrics.py (hunk 4 above)
Comment (AlexDo1) [reply skinfo/metrics.py:405]: Added reference in PR #7

**Pull request merged**